docs/DeveloperGuide.md

```diff
@@ -1,11 +1,233 @@
+---
+layout: default
+title: Developer Guide
+---
+* Table of Contents
+  {:toc}
+
+--------------------------------------------------------------------------------------------------------------------
+
 # Developer Guide
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+References
+1. Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html 
+2. User Guide: https://se-education.org/addressbook-level3/UserGuide.html
+3. 
+
 ## Design & implementation
 
+## Design
+
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" width="280" />
```

Comment:
Perhaps the link doesn't work?
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110757597/e2433198-66be-469c-93f4-bf8719d362d8)